lib/vm.js

```diff
@@ -85,14 +85,38 @@ function validateString(prop, propName) {
                                'string', prop);
 }
 
+function validateBool(prop, propName) {
+  if (prop !== undefined && typeof prop !== 'boolean')
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', propName,
+                               'boolean', prop);
+}
+
+function validateObject(prop, propName) {
+  if (prop !== undefined && (prop === null || typeof prop !== 'object'))
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', propName,
+                               'object', prop);
```

Comment:
It should be `'Object'`. I would also rather combine these validation functions instead of having one for each type.

Reply:
how would i combine them